src/Symfony/Component/VarDumper/Caster/ConstStub.php

```diff
@@ -30,4 +30,27 @@ public function __toString()
     {
         return (string) $this->value;
     }
+
+    /**
+     * Creates ConstStub from bitfield flag.
+     *
+     * @param int    $value  The value of bitfield flag
+     * @param string $prefix The prefix that filters flags
+     *
+     * @return self
+     */
+    public static function fromFlag($value, $prefix)
+    {
+        $constants = get_defined_constants();
+        foreach ($constants as $c => $v) {
+            if ('' !== $prefix && 0 !== strpos($c, $prefix) || ($value & $v) !== $v) {
+                unset($constants[$c]);
+            }
+        }
+
+        asort($constants);
+        $name = implode(' | ', array_flip($constants)) ?: 0;
```

Comment:
you should make sure that the final dumped value is equal to the value of the flag (the flag may add more bits that the defined constants). Otherwise, the original integer must be kept., as the textual representation would not match the dumped value.